osl/interface_linux.go

```diff
@@ -241,8 +241,9 @@ func (n *networkNamespace) AddInterface(srcName, dstPrefix string, options ...If
 	if n.isDefault {
 		i.dstName = i.srcName
 	} else {
-		i.dstName = fmt.Sprintf("%s%d", i.dstName, n.nextIfIndex)
-		n.nextIfIndex++
+		dstNamePrefix := i.dstName
```

Comment:
`i.srcName` is the original name of the veth pipe end which is going to be moved into the container's netns.
`i.dstName` is the new name the above interface will be renamed to, once it is moved.

> dstPrefix is the prefix of the interface in the host system

No,  `dstPrefix` the network driver chosen prefix for the new interface name once we move it into the netns.
`func (n *networkNamespace) AddInterface(srcName, dstPrefix string, options ...IfaceOption) error {`

At this point of the code `i.dstName == dstPrefix` (L229 `i := &nwIface{srcName: srcName, dstName: dstPrefix, ns: n}`)
IMO, original code was already confusing and should have been written as
`i.dstName = fmt.Sprintf("%s%d", dstPrefix, n.nextIfIndex)`